Local eslint is not used

After upgrading to 5.0.0, it seems that my local installation of ESLint is not being used, because I began getting errors that a shared config could not be found.  Installing that config in ~/.atom/packages/linter-eslint prevents the error message.  Downgrading to `4.0.0` also solves the problem.

I'm having this issue as well.

> Error: Cannot find module '@jokeyrhyme/eslint-config-es2015' Referenced from: path/to/my/project/.eslintrc
> Referenced from: path/to/my/project/.eslintrc
>   at Function.Module._resolveFilename (module.js:336:15)
>   at Function.Module._load (module.js:286:25)
>   at Module.require (module.js:365:17)
>    at require (module.js:384:17)
>    at readConfigFromFile (/home/ron/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:155:44)
>    at loadConfig (/home/ron/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:176:22)
>    at /home/ron/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:207:46
>    at Array.reduceRight (native)
>    at loadConfig (/home/ron/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:191:36)
>    at getLocalConfig (/home/ron/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:321:23)

Running `$(npm bin)/eslint .` works fine, so this seems to be a linter-eslint issue.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,10 @@
+### Upcoming
+
+* Support local eslint installation
+
 ### v5.0.1
 
-* Shows a nice notification if the worker process crashed for some reason
+* Shows a nice notification if the worker process crashed for unknown reasons
 
 ### v5.0.0
 

--- a/lib/worker.js
+++ b/lib/worker.js
@@ -11,6 +11,7 @@ const Communication = new CP()
 
 let oldGlobal = null
 let eslintPath = null
+let eslintPathLocal = FS.realpathSync(Path.join(__dirname, '..'))
 let eslint = null
 
 function find(startDir, name) {
@@ -46,26 +47,25 @@ Communication.on('JOB', function(job) {
     modulesPath.pop()
     modulesPath = modulesPath.join(Path.sep)
     process.env.NODE_PATH = modulesPath
-    try {
-      FS.accessSync(eslintNewPath = Path.join(modulesPath, 'eslint'), FS.R_OK)
-    } catch (_) {
-      eslintNewPath = null
-    }
   } else process.env.NODE_PATH = ''
   require('module').Module._initPaths()
 
   process.chdir(params.fileDir)
 
-  if (params.global !== oldGlobal) {
-    eslintNewPath = params.global ?
-      Path.join(
-        ChildProcess.spawnSync('npm', ['get', 'prefix'])
-          .output[1].toString().trim(), 'lib'
-      ) : eslintNewPath || FS.realpathSync(Path.join(__dirname, '..'))
-    oldGlobal = params.global
+  if (params.global) {
+    eslintNewPath = Path.join(
+      ChildProcess.spawnSync('npm', ['get', 'prefix'])
+        .output[1].toString().trim(), 'lib'
+    )
+  } else {
+    try {
+      FS.accessSync(eslintNewPath = Path.join(modulesPath, 'eslint'), FS.R_OK)
+    } catch (_) {
+      eslintNewPath = eslintPathLocal
+    }
   }
 
-  if (eslintNewPath !== null && eslintNewPath !== eslintPath) {
+  if (eslintNewPath !== eslintPath) {
     try {
       eslint = require(Path.join(eslintNewPath, 'node_modules', 'eslint', 'lib', 'cli.js'))
       eslintPath = eslintNewPath

--- a/lib/worker.js
+++ b/lib/worker.js
@@ -13,7 +13,7 @@ let oldGlobal = null
 let oldEslintPath = null
 let eslint = null
 
-function findFile(startDir, name) {
+function find(startDir, name) {
   let filePath
   const chunks = startDir.split(Path.sep)
   while (chunks.length) {
@@ -32,14 +32,14 @@ Communication.on('JOB', function(job) {
   global.__LINTER_RESPONSE = []
 
   if (params.canDisable) {
-    const configFile = findFile(params.fileDir, '.eslintrc')
+    const configFile = find(params.fileDir, '.eslintrc')
     if (configFile === null) {
       job.Response = []
       return
     }
   }
 
-  let modulesPath = findFile(params.fileDir, 'node_modules')
+  let modulesPath = find(params.fileDir, 'node_modules')
   if (modulesPath) {
     modulesPath = modulesPath.split(Path.sep)
     modulesPath.pop()

--- a/lib/worker.js
+++ b/lib/worker.js
@@ -11,7 +11,7 @@ const Communication = new CP()
 
 let oldGlobal = null
 let eslintPath = null
-let eslintPathLocal = FS.realpathSync(Path.join(__dirname, '..'))
+let eslintPathLocal = Path.join(FS.realpathSync(Path.join(__dirname, '..')), 'node_modules', 'eslint')
 let eslint = null
 
 function find(startDir, name) {
@@ -43,9 +43,6 @@ Communication.on('JOB', function(job) {
   let modulesPath = find(params.fileDir, 'node_modules')
   let eslintNewPath = null
   if (modulesPath) {
-    modulesPath = modulesPath.split(Path.sep)
-    modulesPath.pop()
-    modulesPath = modulesPath.join(Path.sep)
     process.env.NODE_PATH = modulesPath
   } else process.env.NODE_PATH = ''
   require('module').Module._initPaths()
@@ -55,7 +52,7 @@ Communication.on('JOB', function(job) {
   if (params.global) {
     eslintNewPath = Path.join(
       ChildProcess.spawnSync('npm', ['get', 'prefix'])
-        .output[1].toString().trim(), 'lib'
+        .output[1].toString().trim(), 'lib', 'node_modules', 'eslint'
     )
   } else {
     try {
@@ -67,7 +64,7 @@ Communication.on('JOB', function(job) {
 
   if (eslintNewPath !== eslintPath) {
     try {
-      eslint = require(Path.join(eslintNewPath, 'node_modules', 'eslint', 'lib', 'cli.js'))
+      eslint = require(Path.join(eslintNewPath, 'lib', 'cli.js'))
       eslintPath = eslintNewPath
     } catch (e) {
       throw new Error('ESLint not found, Please install or make sure Atom is getting $PATH correctly')

--- a/lib/worker.js
+++ b/lib/worker.js
@@ -40,29 +40,38 @@ Communication.on('JOB', function(job) {
   }
 
   let modulesPath = find(params.fileDir, 'node_modules')
+  let eslintNewPath = null
   if (modulesPath) {
     modulesPath = modulesPath.split(Path.sep)
     modulesPath.pop()
-    process.env.NODE_PATH = modulesPath.join(Path.sep)
+    modulesPath = modulesPath.join(Path.sep)
+    process.env.NODE_PATH = modulesPath
+    try {
+      FS.accessSync(eslintNewPath = Path.join(modulesPath, 'eslint'), FS.R_OK)
+    } catch (_) {
+      eslintNewPath = null
+    }
   } else process.env.NODE_PATH = ''
   require('module').Module._initPaths()
 
   process.chdir(params.fileDir)
 
   if (params.global !== oldGlobal) {
+    eslintNewPath = params.global ?
+      Path.join(
+        ChildProcess.spawnSync('npm', ['get', 'prefix'])
+          .output[1].toString().trim(), 'lib'
+      ) : FS.realpathSync(Path.join(__dirname, '..'))
+    oldGlobal = params.global
+  }
+
+  if (eslintNewPath !== null && eslintNewPath !== oldEslintPath) {
     try {
-      const eslintPath = params.global ?
-        Path.join(
-          ChildProcess.spawnSync('npm', ['get', 'prefix'])
-            .output[1].toString().trim(), 'lib'
-        ) :
-        Path.join(__dirname, '..')
-      eslint = require(Path.join(eslintPath, 'node_modules', 'eslint', 'lib', 'cli.js'))
-      oldEslintPath = eslintPath
+      eslint = require(Path.join(eslintNewPath, 'node_modules', 'eslint', 'lib', 'cli.js'))
+      oldEslintPath = eslintNewPath
     } catch (e) {
       throw new Error('ESLint not found, Please install or make sure Atom is getting $PATH correctly')
     }
-    oldGlobal = params.global
   }
 
   job.Response = new Promise(function(resolve) {

--- a/lib/worker.js
+++ b/lib/worker.js
@@ -10,7 +10,7 @@ const FS = require('fs')
 const Communication = new CP()
 
 let oldGlobal = null
-let oldEslintPath = null
+let eslintPath = null
 let eslint = null
 
 function find(startDir, name) {
@@ -65,10 +65,10 @@ Communication.on('JOB', function(job) {
     oldGlobal = params.global
   }
 
-  if (eslintNewPath !== null && eslintNewPath !== oldEslintPath) {
+  if (eslintNewPath !== null && eslintNewPath !== eslintPath) {
     try {
       eslint = require(Path.join(eslintNewPath, 'node_modules', 'eslint', 'lib', 'cli.js'))
-      oldEslintPath = eslintNewPath
+      eslintPath = eslintNewPath
     } catch (e) {
       throw new Error('ESLint not found, Please install or make sure Atom is getting $PATH correctly')
     }
@@ -77,7 +77,7 @@ Communication.on('JOB', function(job) {
   job.Response = new Promise(function(resolve) {
     process.argv = [
       process.execPath,
-      oldEslintPath,
+      eslintPath,
       '--stdin',
       '--format',
       Path.join(__dirname, 'reporter.js'),

--- a/lib/worker.js
+++ b/lib/worker.js
@@ -61,7 +61,7 @@ Communication.on('JOB', function(job) {
       Path.join(
         ChildProcess.spawnSync('npm', ['get', 'prefix'])
           .output[1].toString().trim(), 'lib'
-      ) : FS.realpathSync(Path.join(__dirname, '..'))
+      ) : eslintNewPath || FS.realpathSync(Path.join(__dirname, '..'))
     oldGlobal = params.global
   }
 

